src/Servant/Reflex/Response.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE ConstraintKinds   #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Response handlers
+--   Usefull for 'clientWithOptsAndResultHandler'
+module Servant.Reflex.Response
+  ( reloadOnAPIError
+  , reloadOnResponse
+  , RefreshSupport
+  , isAuthErr
+  , is4xxErr
+  , isDecodeError
+  ) where
+
+import           Control.Monad               (join, void)
+import qualified Data.Text                   as Text
+import           JSDOM                       (currentDocument)
+import           JSDOM.Generated.Document    (getLocation)
+import           JSDOM.Types                 (Location, liftDOM)
+import           Language.Javascript.JSaddle (MonadJSM, ( # ))
+import           Reflex
+import           Reflex.Dom.Builder.Class
+import           Reflex.Dom.Widget.Basic
+import           Reflex.Dom.Xhr
+import           Servant.Reflex
+
+-- | Alias for constraints that can refresh
+type RefreshSupport t m
+   = (MonadJSM m, MonadHold t m, DomBuilder t m, SupportsServantReflex t m)
+
+is4xxErr :: ReqResult () a -> Bool
+is4xxErr =
+  maybe
+    False
+    (\req ->
+       (_xhrResponse_status req < fromInteger 500) &&
+       (_xhrResponse_status req >= 400)) .
+  response
+
+isAuthErr :: ReqResult () a -> Bool
+isAuthErr =
+  maybe
+    False
+    (\req ->
+       let stat = _xhrResponse_status req
+       in (stat == fromInteger 401) &&
```

Comment:
I think you mean `||` here and in the next line, rather than `&&`? (`stat` will never be both 401 and 402)

Reply:
yes probably